x-pack/plugins/reporting/public/share_context_menu/image_export_modal.tsx

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiCopy,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiRadioGroup,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchEvent,
+} from '@elastic/eui';
+import type { IUiSettingsClient, ThemeServiceSetup, ToastsSetup } from '@kbn/core/public';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n-react';
+import url from 'url';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+import React, { FC, useEffect, useState } from 'react';
+import useMountedState from 'react-use/lib/useMountedState';
+import { LayoutParams } from '@kbn/screenshotting-plugin/common';
+import type { JobAppParamsPDFV2 } from '@kbn/reporting-export-types-pdf-common';
+import { ReportingAPIClient } from '../lib/reporting_api_client';
+import { ErrorUrlTooLongPanel, ErrorUnsavedWorkPanel } from './reporting_panel_content/components';
+import { getMaxUrlLength } from './reporting_panel_content/constants';
+import type { JobParamsProviderOptions } from '.';
+import { AppParams } from '../lib/reporting_api_client/reporting_api_client';
+
+export interface ReportingModalProps {
+  apiClient: ReportingAPIClient;
+  toasts: ToastsSetup;
+  uiSettings: IUiSettingsClient;
+  reportType?: string;
+  requiresSavedState: boolean; // Whether the report to be generated requires saved state that is not captured in the URL submitted to the report generator.
+  objectId?: string;
+  isDirty?: boolean;
+  onClose: () => void;
+  theme: ThemeServiceSetup;
+  layoutOption?: 'print' | 'canvas';
+  // canvas breaks if required
+  jobProviderOptions?: JobParamsProviderOptions;
+  // needed for canvas
+  getJobParams?: JobAppParamsPDFV2;
+}
+
+export type Props = ReportingModalProps & { intl?: InjectedIntl };
+
+export const ReportingModalContentUI: FC<Props> = (props: Props) => {
+  const { apiClient, intl, toasts, theme, onClose, objectId, layoutOption, jobProviderOptions } =
+    props;
+
+  const isSaved = Boolean(objectId);
+  const [isStale, setIsStale] = useState(false);
+  const [createReportingJob, setCreatingReportJob] = useState(false);
+  const [selectedRadio, setSelectedRadio] = useState<'pngV2' | 'printablePdfV2' | 'printablePdf'>(
+    'printablePdfV2'
+  );
+  const [usePrintLayout, setPrintLayout] = useState(false);
+  const [useCanvasLayout, setCanvasLayout] = useState(false);
+  const [absoluteUrl, setAbsoluteUrl] = useState('');
+  const isMounted = useMountedState();
+  const [objectType] = useState<string>('dashboard');
```

Comment:
I'm noticing when I create a PDF report in Lens, a toast message appears saying `Queued report for dashboard`. It should say "report for lens". It must be because this value never changes. We will need use the `setObjectType` function that comes from useState.

Reply:
Fixed in 186d3f1 thank you for your help!